Extracted MTU check code into its own helper functions.

Dimi1010 opened the pull request:
This PR separates the logic for checking packets against the device MTU size by moving validation code to `isPayloadWithinMtu` helper functions.

The change simplifies the control flow of `sendPacket` by changing the behaviour from:
```
1. sendPacket(uint8_t* data, size_t dataLen, true /* checkMtu */)
  1.1. sendPacket(RawPacket&, true /* checkMtu */)
    1.1.1. sendPacket(Packet&, true /* checkMtu */)
      1.1.1.1. doMtuCheck(/* ... */) // does check
      1.1.1.2. sendPacket(RawPacket&, false /* no mtu check */)
        1.1.1.2.1. sendPacket(uint8_t* data, size_t dataLen, false /* no mtu check */)
          1.1.1.2.1.1. sendPacketDirect(/* ... */) // does transmission
```
to:
```
1. sendPacket(uint8_t* data, size_t dataLen, true /* checkMtu */)
  1.1. isPayloadWithinMtu(uint8_t* data, size_t dataLen)
    1.1.1. isPayloadWithinMtu(RawPacket&)
      1.1.1.1. isPayloadWithinMtu(Packet&) // does check
  1.2. sendPacketUnchecked(/* ... */) // does transmission
```

It also renames `sendPacketDirect` to `sendPacketUnchecked` to improve readability.

Dimi1010 commented on Pcap++/header/PcapLiveDevice.h:
```diff
@@ -637,8 +637,55 @@ namespace pcpp
 	protected:
 		internal::PcapHandle doOpen(const DeviceConfiguration& config);
 
+		/// @brief Checks whether the packetPayloadLength is larger than the device MTU.
```
```suggestion
		/// @brief Checks whether the packetPayloadLength is smaller or equal than the device MTU.
```

Dimi1010 commented on Pcap++/header/PcapLiveDevice.h:
```diff
@@ -637,8 +637,55 @@ namespace pcpp
+		/// @return True if the payloadLength is less than or equal to the device MTU
+		bool checkMtu(size_t payloadLength) const;
+
+		/// @brief Checks whether the packet's payload length is larger than the device MTU.
```
```suggestion
		/// @brief Checks whether the packet's payload length is smaller or equal than the device MTU.
```

Dimi1010 commented on Pcap++/header/PcapLiveDevice.h:
```diff
@@ -637,8 +637,55 @@ namespace pcpp
+		/// @return True if the packet's payload length is less than or equal to the device MTU.
+		bool checkMtu(Packet const& packet, bool allowUnknown = true, size_t* outPayloadLength = nullptr) const;
+
+		/// @brief Checks whether the payload length of a RawPacket is larger than the device MTU.
```
```suggestion
		/// @brief Checks whether the payload length of a RawPacket is smaller or equal than the device MTU.
```

Dimi1010 commented on Pcap++/header/PcapLiveDevice.h:
```diff
@@ -637,8 +637,55 @@ namespace pcpp
+		/// @return True if the packet's payload length is less than or equal to the device MTU.
+		bool checkMtu(RawPacket const& rawPacket, bool allowUnknown = true, size_t* outPayloadLength = nullptr) const;
+
+		/// @brief Checks whether the payload length of a packet's raw data is larger than the device MTU.
```
```suggestion
		/// @brief Checks whether the payload length of a packet's raw data is smaller or equal than the device MTU.
```

Dimi1010 commented on Pcap++/header/PcapLiveDevice.h:
```diff
@@ -637,8 +637,55 @@ namespace pcpp
+		/// If allowUnknown is set to true, packets which's length cannot be determined will be allowed.
+		/// In such cases if outPayloadLength is not nullptr, it will be set to the maximum possible payload length.
+		///
+		/// @param packet The packet to check
+		/// @param allowUnknown Controls whether packets with unknown payload length are allowed.
+		/// @param outPayloadLength If not nullptr, the payload length of the packet will be written to this pointer.
```
```suggestion
		/// @param[in] packet The packet to check
		/// @param[in] allowUnknown Controls whether packets with unknown payload length are allowed.
		/// @param[out] outPayloadLength If not nullptr, the payload length of the packet will be written to this pointer.
```

Dimi1010 reviewed (commented):
Comment fixes.

seladb commented:
Having both `doMtuCheck()` as a public method and `checkMtu()` as a protected method is even more confusing IMO. I'm ok with removing the error log in `doMtuCheck()` if you think it's a cleaner API

Dimi1010 commented:
@seladb Changed the function name to be more explicit and moved the new functions to public API.
Also added a new section to the PR post explaining how the change improves readability by simplifying the control flow of the device internals.

seladb commented:
@Dimi1010 I'm still not convinced this change is needed. We add multiple public methods to `PcapLiveDevice` which makes the API more complex for no apparent reason. Also - the existing logic (before this change) is clear and readable enough and I don't think we need this refactoring

Dimi1010 commented:
@seladb moved the declarations to protected, or do you want them explicitly in private?
